bgc_derivation/derivations/oxygen/DOXY_201_202_202.m

```diff
@@ -11,30 +15,38 @@
             % derivation.compute_parameter()
             %
             
-            % Fetch all required parameters from source data
-
-            bphasedoxy = self.variables('BPHASE_DOXY');
-            rphasedoxy = self.variables('RPHASE_DOXY');
-            pres = self.variables('PRES');
-            temp = self.variables('TEMP');
-            psal = self.variables('PSAL');
+            % Fetch all required parameters from source data/metdata
+            pres = self.variables.p;
+            temp = self.variables.t;
+            psal = self.variables.s;
+            bphase = self.variables.TPhase;
 
             % Any adjustments or transformations applied to source data
             % to meet expectations of the underlying eqaution should happen
             % here.
-
-		            
-            % Compute desired output value(s)
-            doxy = self.equation(bphasedoxy, rphasedoxy, pres, temp, psal,self.coeffs );
+            pres = force_row(pres);
+            temp = force_row(temp);
+            psal = force_row(psal);
+            bphase = force_row(bphase);
+            rphase = zeros(size(bphase));
+            
+            % Final checks
             if isempty(self.coeffs) 
                 error('Error: calibration coefficients not set');
-            else
+            end
+		            
+            % Compute desired output value(s)
+            doxy = self.equation(bphase, rphase, pres, temp, psal, self.coeffs);
+            
             % Any subsequent adjustments or transformations for consistency
             % should happen here.
+            if any(size(pres) ~= size(self.variables.p))
```

Comment:
Why particularly pres and not the other vars? can we assume if one var is fine everything else will be?

Reply:
It could be any source data variable. If they're not all the same size, the equation code would have errored out before we reach this line.